joeynmt/prediction.py

```diff
@@ -63,6 +63,11 @@ def validate_on_data(model: Model, data: Dataset,
         - decoded_valid: raw validation hypotheses (before post-processing),
         - valid_attention_scores: attention scores for validation hypotheses
     """
+    if batch_size > 1000 and batch_type == "sentence":
+        print("WARNING: Are you sure you meant to work on huge batches like "
```

Comment:
Maybe use logger.warning instead of print ?

Reply:
At that point there's no access to a logger... Since it's possibly outside training and we don't use a logger during testing. Will add one for all testing/translating modes as well.